R/insertPackage.R

```diff
@@ -246,14 +256,31 @@ getPackageInfo <- function(file) {
 }
 
 contrib.url2 <- function(repos, type = getOption("pkgType"), version = NULL){
-    contrib_url <- contrib.url(repos = repos, type = type)
-    if(is.null(version) || is.na(version)){
-        return(contrib_url)
+    FUN <- function(t){
+        contrib_url <- contrib.url(repos = repos, type = t)
+        if(is.null(version)){
+            return(contrib_url)
+        } else if(is.na(version)) {
+            if(t != "source"){
+                contrib_url <- c(contrib_url,
+                                 list.dirs(gsub(DRAT_CONTRIB_VERSION_REGEX, "contrib",
+                                             contrib_url),
+                                        recursive = FALSE))
+                contrib_url <- unique(contrib_url)
+            }
+        } else {
+            version <- package_version(version)
+            contrib_url <- gsub(DRAT_CONTRIB_VERSION_REGEX,
+                                paste0("contrib/",
+                                       paste0(version$major,".",version$minor)),
+                                contrib_url)
+            contrib_url
+        }
+        contrib_url
     }
-    version <- package_version(version)
-    contrib_url <- gsub("contrib/[0-9]\\.[0-9]",
-                        paste0("contrib/",
-                               paste0(version$major,".",version$minor)),
-                        contrib_url)
-    contrib_url
+    urls <- lapply(type,FUN)
+    names <- unlist(mapply(rep,type,lengths(urls)))
```

Comment:
I haven't investigate inputs here, but if you `unlist` here anyway, then safer to use `mapply(SIMPLIFY=FALSE)`

Reply:
fixed in 8a15269